### ISSUE TITLE
markdown link validate not supporting MDX

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#4217]
Does this issue occur when all extensions are disabled?: Yes

-- Version & env

- VS Code Version: : 1.69.1
- OS Version: Darwin arm64 21.3.0
Commit: b06ae3b2d2dbfe28bca3134cc6be65935cdfea6a
Date: 2022-07-12T08:22:10.822Z (5 days ago)
Electron: 18.3.5
Chromium: 100.0.4896.160
Node.js: 16.13.2
V8: 10.0.139.17-electron.0

Also tested with latest insider version with minimal settings:
```json
{
    "files.associations": {
        "*.mdx": "markdown"
    },
    "markdown.experimental.validate.enabled": true,
}
```
Version: 1.70.0-insider
Commit: 1cd90cceddf3c413673963ab6f154d2ff294b17c
Date: 2022-07-15T05:32:19.076Z
Electron: 18.3.5
Chromium: 100.0.4896.160
Node.js: 16.13.2
V8: 10.0.139.17-electron.0
OS: Darwin arm64 21.3.0

-- Repro

MRE:

I've created a MRE here https://github-com/PabloLION/issue-vscode-mdx-validate
Open any `.md` or `.mdx` file.
There's a warning on the third line.

Steps to Reproduce:

1. Set "*.mdx" file to associate with markdown. 
2. Enable `markdown.experimental.validate.enabled` by adding a `"markdown.experimental.validate.enabled": true,` to vscode settings.
3. Add a link to any `mdx` file (create one if not exists) without extension
4. A warning of "File does not exist at path: ..." was raised
5. Try to open the link. The file without extension does not exist showing "The editor could not be opened because the file was not found. [Create File]"

-- TLDR

- Why I think this is a bug
If `mdx` is set as an alternative extension of `markdown`, all `.mdx` files should be treated same as `.md`. But in this case of validating, the `.mdx` files cannot be detected.  
Maybe we can consider 
1. adding `*.mdx` as a fallback of `*.md` when the latter one doesn't exists.
2. adding a setting for "default extension for relative link" and report warning when too many files are found.
3. let `.mdx` link to look for `.mdx` files only. (this seems should be done with an extension)
